### PR TITLE
Fix #633: Add PrimeFaces.current() as injectable bean.

### LINKED ISSUE #633: Injectable PrimeFaces
PrimeFaces right now is a static utility similar to FacesContext you use it like PrimeFaces.current();

Submitting a PR to make it Autowirable to make injection and unit testing easier.

This will allow

```java
   @Autowired
   PrimeFaces primefaces;
```

### PR DESCRIPTION
**Pull request by melloware**
PrimeFaces right now is a static utility similar to FacesContext you use it like PrimeFaces.current();

Submitting a PR to make it Autowirable to make injection and unit testing easier.

This will allow

```java
   @Autowired
   PrimeFaces primefaces;
```

### CODE REVIEW

**Comment by larsgrefer** [comment]
@melloware Thanks for your contribution.

Please fix the checkstyle issues: https://travis-ci.org/joinfaces/joinfaces/jobs/480568744#L1700-L1712

**Comment by melloware** [comment]
Sorry. I will fix

File: joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java [hunk 1]
```diff
@@ -88,4 +94,11 @@
 	@EnableConfigurationProperties(Primefaces6_2Properties.class)
 	public static class Primefaces6_2AutoConfiguration {
 	}
+	
+	@Bean("primefaces")
+	@ConditionalOnMissingBean(name = "primefaces")
+	@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE, proxyMode = ScopedProxyMode.TARGET_CLASS)
```
Comment (larsgrefer) [inline joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java]: `PrimeFaces.getInstance()` seems to return the very same object instance on every invocation, regardless of the current request or session context. Therefore `prototype` scope seems unnecessary.

Comment (melloware) [reply joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java]: @larsgrefer I followed the exact pattern of JsfAutoConfiguraton which returns FacesContext.getCurrentInstance() the same exact way as a prototype and not proxied.

File: joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java [hunk 2]
```diff
@@ -88,4 +94,11 @@
 	@EnableConfigurationProperties(Primefaces6_2Properties.class)
 	public static class Primefaces6_2AutoConfiguration {
 	}
+	
+	@Bean("primefaces")
+	@ConditionalOnMissingBean(name = "primefaces")
```
[the remark this answers is not in the document]
Comment (melloware) [reply joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java]: fixed to look for (PrimeFaces.class) now

[the remark this answers is not in the document]
Comment (melloware) [reply joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java]: Fixed per your suggestion

File: joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java (hunk 1 above)
Comment (melloware) [reply joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java]:
```java
	@Bean("facesContext")
	@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE, proxyMode = ScopedProxyMode.TARGET_CLASS)
	@ConditionalOnMissingBean
	public FacesContext facesContext() {
		return FacesContext.getCurrentInstance();
	}
```

Comment (larsgrefer) [reply joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java]: But `FacesContext.getCurrentInstance()` returns different object instances depending on the context (thread) its called in. Therefore the corresponding spring bean uses the prototype scope, so every time it's used `FacesContext.getCurrentInstance()` is invoked to obtain the correct `FacesContext` instance (and not the instance from 5 requests ago).

Comment (melloware) [reply joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/primefaces/PrimefacesAutoConfiguration.java]: Understood. I just removed the Scope annotation.

**Pull request merged**